workflow/controller/controller.go

```diff
@@ -512,7 +512,18 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 		pods := wfc.kubeclientset.CoreV1().Pods(namespace)
 		switch action {
 		case terminateContainers:
-			if terminationGracePeriod, err := wfc.signalContainers(namespace, podName, syscall.SIGTERM); err != nil {
+			pod, err := wfc.getPod(namespace, podName)
+			if err == nil && pod != nil && pod.Status.Phase == "Pending" {
+				logCtx.WithFields(log.Fields{"name": podName, "namespace": namespace}).Info("deleting pending pod")
+				propagation := metav1.DeletePropagationBackground
+				err = pods.Delete(ctx, podName, metav1.DeleteOptions{
```

Comment:
Can we use the Pod cleanup queue delete it?

Reply:
Ah, not something I've come across yet. It sounds cleaner.

Reply:
> Can we use the Pod cleanup queue to delete it?
OK, I have requeue it to cleanup queue to delete it